postgres/datadog_checks/postgres/statements.py

```diff
@@ -36,6 +41,7 @@
          ON pg_stat_statements.dbid = pg_database.oid
   WHERE query != '<insufficient privilege>'
   AND query NOT LIKE 'EXPLAIN %%'
+  AND queryid = ANY(ARRAY[{called_queryids}]::bigint[])
```

Comment:
The pattern `ARRAY[{called_queryids}]` is gonna be an issue I think.

With 2 ids, you will have `ARRAY([id1, id2])`
With 3 ids, you will have `ARRAY([id1, id2, id3])`

Those queries will be tracked by pgss itself. Every time you have a new parameter, this is tracked as a new query with a new query_id.

Testing quickly
```
sql_string = 'SELECT * FROM pgbench_accounts WHERE aid = ANY(ARRAY[{aids}])'

cur.execute(sql_string.format(aids=[1, 2, 3]))
cur.execute(sql_string.format(aids=[1, 2, 3, 4]))
```

Looking at the result:
```
select queryid, query from pg_stat_statements;
       queryid        |                                  query
----------------------+-------------------------------------------------------------------------
  1123605973931390435 | BEGIN
  -359344131479016271 | select * from pg_stat_statements_reset()
  2010392662945087733 | select * from pg_stat_statements
 -7253646950380962652 | SELECT * FROM pgbench_accounts WHERE aid = ANY(ARRAY[[$1, $2, $3, $4]])
  7524854806514990763 | SELECT * FROM pgbench_accounts WHERE aid = ANY(ARRAY[[$1, $2, $3]])
```

A way to avoid this is to pass parameters as a literal array `'{id1, id2, id3...}'`.

```
sql_string_2 = "SELECT * FROM pgbench_accounts WHERE aid = ANY('{{ {} }}')"
cur.execute(sql_string_2.format(','.join([str(a) for a in [1, 2, 3, 4]])))
cur.execute(sql_string_2.format(','.join([str(a) for a in [1, 2, 3, 4, 5]])))
```

This time, there's only a single tracked query:
```
select queryid, query from pg_stat_statements;
       queryid        |                       query
----------------------+----------------------------------------------------
 -2815121331412783721 | select queryid, query from pg_stat_statements
  1123605973931390435 | BEGIN
  -359344131479016271 | select * from pg_stat_statements_reset()
 -3011367731889348821 | SELECT * FROM pgbench_accounts WHERE aid = ANY($1)
```

Reply:
One `pg_stat_statement` row is definitely better. I'm testing out that change now.